### PR TITLE
RigidBody Vertex Editor

### PR DESCRIPTION
**Pull request by YMRYMR**
### Pull Request Checklist

_This is a reminder on a few important traits of your Pull Request. **Do not add this to your description**._

- Have you read the [**wiki article** on contribution](https://github.com/AdamsLair/duality/wiki/How-to-Contribute)? : Yes
- Have you done your changes in a **PR-specific branch**? If not, create one and pull your changes into it before submitting. In case your PR gets rejected, your `master` branch won't be out of sync. : What? Sorry. No idea. Not used to GitHub. Should I? How?
- Have you checked **spaces and tabs**? Use tabs for indentation and spaces for alignment. : Yes
- Does your **code style** match Duality code style? : Not sure
- Have you **documented public API** you added or changed? : No
- Have you **tested thoroughly** forfunctionality, stability and usability? : Yes
- Do all of your changes and additions comply with the **comply with the [project license](https://github.com/AdamsLair/duality/blob/master/LICENSE)**? : Yes

### CODE REVIEW

**Comment by YMRYMR** [comment]
Just learned how to create a feature branch... Too late I guess. Sorry about that.

**Comment by ilexp** [comment]
It seems like something in a find / replace all operation went wrong - there are 245 changed files, which is definitely much more than there should be. Before we get to the review part, can you clean up your PR by reverting all files that do not necessarily need to be changed for this?

Also, from the looks of the diffs, it seems like you replaced all "four spaces" occurrences with a tab across the entire solution. However, this created problems wherever whitespace was used for alignment reasons - please make sure to revert these cases as well. In general, we're using tabs for indentation, but spaces for alignment. A typical example is the field definition block of a class:

```CSharp
	private string fieldOne   = "Hello";
	private int    fieldTwo   = 42;
	private float  fieldThree = 3.14f;
```

Note that whitespace in front of the first character is all tabs - that's the indentation part. It allows different users to view the code at different tab / indentation sizes. However, the whitespace that is used to make sure the field names and values align properly consists of spaces - this ensures that the alignment is kept across all tab sizes.

In short: Keep all spaces that are not leading whitespace. Use tabs otherwise. Not sure if that rule is true 100% of the time across corner cases, but you get the idea :)

> Just learned how to create a feature branch... Too late I guess. Sorry about that.

Np. Next time!

**Comment by afraidOfMight** [comment]
Small tip: If you are using Strg + K Strg + D (Autoformatting) you have to select "Ignore spacing"(Don't know the exact wording in english) in the options: Extras->Options->TextEditor->C#->(Formatting)->(Spacing)->(Other spacing options)-> (Ignore spaces around declarations)
Not so sure about the wording within ()
Hope it helps.

**Comment by YMRYMR** [comment]
Thanks, I'll try that.
Didn't find the rollback option yet, but made a list of the modified files so I don't miss any of the changes:
> Source/Plugins/EditorModules/CamView/CamView.Editor.csproj
> Source/Plugins/EditorModules/CamView/CamViewLayers/RigidBodyShapeCamViewLayer.cs
> .../EditorModules/CamView/CamViewStates/RigidBodyEditor/SelObj/RigidBodyEditorSelVertices.cs
> 
> ...torModules/CamView/CamViewStates/RigidBodyEditor/Overlay/PolygonRigidBodyEditorOverlay.cs
> ...lugins/EditorModules/CamView/CamViewStates/RigidBodyEditor/RigidBodyEditorCamViewState.cs
> Source/Plugins/EditorModules/CamView/CamViewStates/RigidBodyEditor/RigidBodyEditorTool.cs
> ...lugins/EditorModules/CamView/CamViewStates/RigidBodyEditor/Tools/NoRigidBodyEditorTool.cs
> .../EditorModules/CamView/CamViewStates/RigidBodyEditor/Tools/PolyLikeRigidBodyEditorTool.cs
> ...s/EditorModules/CamView/CamViewStates/RigidBodyEditor/Tools/PolygonRigidBodyEditorTool.cs
> 
> ...ns/EditorModules/CamView/CamViewStates/RigidBodyEditor/IRigidBodyEditorToolEnvironment.cs
> ...ns/EditorModules/CamView/CamViewStates/RigidBodyEditor/Tools/VertexRigidBodyEditorTool.cs
> Source/Plugins/EditorModules/CamView/Properties/CamViewRes.Designer.cs
> Source/Plugins/EditorModules/CamView/Properties/CamViewRes.resx
> Source/Plugins/EditorModules/CamView/Properties/CamViewResCache.cs
> Source/Plugins/EditorModules/CamView/Resources/curArrowEditVertices.png
> Source/Plugins/EditorModules/CamView/Resources/iconCmpEditVertices.png
> Source/Plugins/EditorModules/CamView/UndoRedoActions/EditRigidBodyPolyShapeAction.cs
> Source/Plugins/EditorModules/CamView/UndoRedoActions/RigidBodyShapeVertexAction.cs
> 
> ...ns/EditorModules/CamView/CamViewStates/RigidBodyEditor/Tools/CircleRigidBodyEditorTool.cs
> .../EditorModules/CamView/CamViewStates/RigidBodyEditor/Tools/PolyLikeRigidBodyEditorTool.cs

**Comment by ilexp** [comment]
Used my push access that was granted via this PR to clean up some of the tabs / spaces issues myself and revert the commit that globally replaced things. Now we're at 19 changed files, which is much better. Will try to get some time to dive into the code and either do a review for next iteration, or do some changes myself. 

Right now it looks like the last option will be the case, since @YMRYMR told me that he might be offline for a while, or at least not in a position to continue developing on this PR.
